### ISSUE TITLE
🐛 BUG: Getting 404 on URL that contains a dot and no trailing slash

### ISSUE THREAD

**Issue by rcdailey** [cloudflare/workers-sdk#2779]
### Which Cloudflare product(s) does this pertain to?

Pages

### What version of `Wrangler` are you using?

`cloudflare/pages-action@v1`

### What operating system are you using?

Github Actions

### Describe the Bug

I [originally posted about this](https://community.cloudflare.com/t/getting-404-on-url-that-contains-a-dot-and-no-trailing-slash/476343?u=voidpointer) in the Cloudflare community forum.

I host my Docusaurus documentation site on Cloudflare Pages, and I use Algolia Crawl to index the site content for my search functionality.

If you do a `GET` on https://recyclarr.dev/wiki/upgrade-guide/v5.0, you will see it returns a 404. If I add a slash to the end, I get a 200. Weirdly, even though it returns 404 the page is still served and rendered correctly in my browser. However, a non-200 response breaks Algolia Crawl.

I suspect that the `.` character in the URL is causing an issue, because I do not get 404 responses for other pages without the dot. To me, this indicates a bug that should be fixed. I expect that the URL linked above should return 200 without a trailing slash at the end.

### Additional Information

My docs site is open source and may be located here: https://github.com/recyclarr/wiki

I do not use Wrangler or anything. I build and deploy my site using a [Github Workflow](https://github.com/recyclarr/wiki/blob/master/.github/workflows/build-and-deploy.yml)

**Comment by erictor** [comment]
Very interested to see if you get response and info on this.  Seeing the same issue.  It doesn't happen in development in the local dev server environment, but only after publishing.  What might be happening is an assumption, somewhere, that if there is a period in the last segment of a URI that it must be a request for a file rather than being a route.  If that's what's occurring, I'd like to think there ought to be a way to set a setting to stop that behavior.

**Comment by JacobMGEvans** [comment]
@cloudflare/pages should this be in the `cloudflare/pages-action` repo?

**Comment by GregBrimble** [comment]
Here's probably the most appropriate, actually. If it's in any open-source code, it'll be pages-shared. But given the report suggesting that it's only affecting production, it may be something that's internal. We'll keep this issue here and take a look.

**Comment by JacobMGEvans** [comment]
Thanks @GregBrimble 😃

**Comment by rcdailey** [comment]
Hey folks. Any idea where this one falls on the roadmap / priority list? I'm guessing the affected audience for this one is low. Let me know if there's anything I can do to help get to a resolution.

**Comment by erictor** [comment]
MS PLEASE give us a way to allow for this, and furthermore, allow for it by setting a config such as in web.config, so that one could do it if one was on a hosting plan and did not have direct access to IIS host settings.   For example, on Instagram they allow usernames that include a period within, like my.username, and the user's account becomes accessible with instagram.com/my.username.  I would like to implement similar functionality and when I tried to use a Regex in web.config to allow for this when the request is not a file but it failed and caused other worse issues to occur.

**Comment by rcdailey** [comment]
@1000hz thank you so much for the fix! How will I know when the fix is live/available so I can tests for the issue on my site?

**Comment by erictor** [comment]
Hi Cina Saffary,

I’m still getting HTTP 404 on listjar.com, unless I append a / to the end of this URI.

***@***.***

Thank you,

Eric
erictor | ***@***.******@***.***>

**Comment by erictor** [comment]
Hi.  So, once again, I have NO idea whether my intrusion on Issue 2779 is relevant or appropriate as I am not a user of workers-sdk.  I do have Cloudflare but I have zero knowledge whether Cloudflare is any part of my problem or can provide a solution.  But I was hoping maybe someone on this Issue thread could give me a hint as to what I could do.  I wanted to mention that after getting notification that this Issue is considered resolved that I tested my particular problem and it remains.  (See attached image).  If anyone has any thoughts I'd love to hear them.  Thank you very much.
![backslash-needed-with-period](https://github.com/cloudflare/workers-sdk/assets/13494060/302366ea-fe8b-4c63-a6f7-3fcf23b63ee1)

**Comment by WalshyDev** [comment]
The next Wrangler release will have the fix @erictor @rcdailey 

Not sure exactly when that will be but keep an eye out for a Wrangler update :)

**Comment by rcdailey** [comment]
@WalshyDev So if I watch this repo for Releases, that'll tell me when this is available?

https://github.com/cloudflare/workers-sdk

**Comment by WalshyDev** [comment]
Yep!

**Comment by hurrellit** [comment]
I am still seeing this behaviour as explained here: https://community.cloudflare.com/t/404-on-urls-containing-dots/599220 - does anyone know if the fix in https://github.com/cloudflare/workers-sdk/pull/4051 has indeed made it to "live" yet?

**Comment by hurrellit** [comment]
This is still happening.... I guess [#4501](https://github.com/cloudflare/workers-sdk/pull/4051) is live but it doesn't fix my issue....

For reference:

https://doc-6ly.pages.dev/technologies/asp/extensions/com.castsoftware.asp >>> 404
https://doc-6ly.pages.dev/technologies/asp/extensions/com.castsoftware.asp/ >>> OK

It doesn't happen in other providers (e.g. Gitlab pages) so this is a Cloudflare specific issue...

**Comment by erictor** [comment]
I have a web site that uses IIS on a hosted platform and is fronted by Cloudflare.  I just tested this on 2024-04-09.  I have a user account for my website registered on my website for testing purposes, with the username "my.name".  Right now, if I browse to mywebsite.com/my.name/ the route successfully displays the profile of the my.name user.  If I browse to mywebsite.com/my.name without the ending-slash, an HTTP ERROR 404 occurs.

**Comment by erictor** [comment]
I should add that I have other 'registered users' on the site with usernames that don't have a period in the username and they work fine with or without an ending-slash.  I'm not so sure this is a 'Cloudflare-specific issue' with regards to my website, I'm thinking it's an issue with IIS, and those who have direct access to the IIS server can configure something to solve this.  However I'm using shared hosting and I don't have the ability to specifically configure IIS.  So, I want Microsoft to give me a solution through web.config (they don't appear to) or I was hoping perhaps Cloudflare could provide an answer to this -- like maybe let me set a configuration to automatically append a slash to the end of a URL that has a period in it, before forwarding the request on to my shared hosting that uses IIS.

**Comment by hurrellit** [comment]
I'm not sure when, but it seems that my problem described [above](https://github.com/cloudflare/workers-sdk/issues/2779#issuecomment-2044648654) has been fixed. I no longer get 404s on URLs containing dots where the trailing slash is not used, instead the URL is redirected and a trailing slash is added, which is the expected behaviour. Thanks if somebody did something....

**Comment by erictor** [comment]
@hurrellit I excitedly went to my site and put in the /my.name user and........... nope.  Same problem.   /my.name/ works, /my.name doesn't.   Maybe they fixed it for specifically your site?

**Comment by hurrellit** [comment]
Maybe post up a link to your site so we can see?  Your first post above talks about web.config and Microsoft IIS web server..?

**Comment by WalshyDev** [comment]
This bug should have been fixed long ago, if you're still having issues please open a new issue.

**Comment by erictor** [comment]
@WalshyDev It appears a solution in the technology my site uses (.NET / IIS) can be URL Rewrite in IIS Manager - using that, one can specify a `<match url="(.*[^/])" />` i.e. a path that's got a dot near the end that doesn't end with a forward slash, and then rewrite the URL to end with the slash before processing.   But my site is on a shared host; I've tried some stuff using web.config and what I've tried has failed.
